database/insertData.js

```diff
@@ -365,17 +391,58 @@ const queries = [
             ]
           }
           templateActionsUsingId: {
-            create: {
-              actionCode: "cLog"
-              condition: { value: true }
-              trigger: ON_APPLICATION_SUBMIT
-              parameterQueries: {
-                message: {
-                  value: "The Action has been executed. Automated Actions FTW!!!"
+            create: [
+              {
+                actionCode: "cLog"
+                condition: { value: true }
+                trigger: ON_APPLICATION_SUBMIT
+                parameterQueries: {
+                  message: { value: "Company Registration submission" }
                 }
               }
-            }
-            # TO-DO: Create actions to add Org, etc.
+              {
+                actionCode: "incrementStage"
+                condition: { value: true }
+                trigger: ON_APPLICATION_CREATE
+                parameterQueries: {
+                  applicationId: {
+                    operator: "objectProperties"
+                    children: [{ value: { property: "record_id" } }]
+                  }
+                }
+              }
+              {
+                actionCode: "changeStatus"
+                condition: { value: true }
+                trigger: ON_APPLICATION_SUBMIT
+                parameterQueries: {
+                  applicationId: {
+                    operator: "objectProperties"
+                    children: [{ value: { property: "record_id" } }]
+                  }
+                  newStatus: { value: "Submitted" }
+                }
+              }
+              {
+                actionCode: "incrementStage"
+                condition: { value: true }
+                trigger: ON_REVIEW_SAVE
+                parameterQueries: {
+                  applicationId: {
+                    type: "number"
+                    operator: "pgSQL"
+                    children: [
+                      { value: "SELECT application_id FROM review WHERE id = $1" }
+                      {
+                        operator: "objectProperties"
+                        children: [{ value: { property: "record_id" } }]
+                      }
+                    ]
+                  }
+                }
+              }
```

Comment:
Just another example, really, for testing the actions work fine with different triggers on different tables.

The query is different, because from the Review table the `record_id` is not the `application_id`, so it needs to do a bit more logic to deduce the application id. I'm actually thinking it might be a good idea to add some additional functionality to the triggers so that `application_id` and `template_id` (and maybe some other values?) always get passed to the Trigger Processor in the notification payload, to avoid having to do these additional lookups for common values.

Have made a card in the backlog: https://github.com/orgs/openmsupply/projects/6#card-48677106

Reply:
Have actually done this #96, with quite a bit of extra documentation, which should make it easier to follow.